src/websocket.js

```diff
@@ -182,11 +168,37 @@ const trades = (payload, cb) => {
         tradeId,
       })
     })
-
-    return w
   })
 
-  return () => cache.forEach(w => w.close())
+  return () => cache.forEach(close => close())
+}
+
+
+const createSocket = (path, cb) => {
+  const onClose = ()=> {
+    if (!closedManually) {
+      w = newSocket(path, cb, onClose)
+    }
+  }
+  
+  let closedManually = false
```

Comment:
Not a fan of this hacky flag. Shouldn't we check the `code` of the event emitted on close and infer if it has been triggered manually or not? Referring to [section 7.4.1](https://tools.ietf.org/html/rfc6455#section-7.4.1) of RFC6455.

Reply:
Neither am I, and initially I was implementing this with using a code but after seeing the [amount](https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent) of possible values for the `code` we need to check + the uncertainty of what codes we'll get working with binance's unstable servers + the general instability of websockets api + the difficulty of testing this + lack of time in my hands I see this simple approach as much more efficient, simpler, and easier to understand than trying to be spec compliant